geopandas/geodataframe.py

```diff
@@ -118,6 +118,15 @@ def __init__(self, *args, geometry=None, crs=None, **kwargs):
         # allowed in that case
         # TODO do we want to raise / return normal DataFrame in this case?
         if geometry is None and "geometry" in self.columns:
+            # Check for multiple columns with name "geometry". If there are,
+            # self["geometry"] is a gdf and constructor gets recursively recalled
+            # by pandas internals trying to access this
+            if len(self.columns[self.columns == "geometry"]) > 1:
```

Comment:
```suggestion
            if (self.columns == "geometry").sum() > 1:
```

(that doesn't need to actually create the filtered Index object to check the length)